### ISSUE TITLE
Flutter 3.0 build error

### ISSUE THREAD

**Issue by anisalibegic** [juliansteenbakker/mobile_scanner#127]
After upgrading to Flutter 3.0 we got a build problem when using mobile_scanner.

Log:
```
../../../SDK/Flutter/.pub-cache/hosted/pub.dartlang.org/mobile_scanner-1.0.0/lib/src/mobile_scanner.dart:50:20: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../../SDK/Flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance?.addObserver(this);
                   ^
../../../SDK/Flutter/.pub-cache/hosted/pub.dartlang.org/mobile_scanner-1.0.0/lib/src/mobile_scanner.dart:131:20: Warning: Operand of null-aware operation '?.' has type 'WidgetsBinding' which excludes null.
 - 'WidgetsBinding' is from 'package:flutter/src/widgets/binding.dart' ('../../../SDK/Flutter/packages/flutter/lib/src/widgets/binding.dart').
    WidgetsBinding.instance?.removeObserver(this);
                   ^
e: D:\Development\SDK\Flutter\.pub-cache\hosted\pub.dartlang.org\mobile_scanner-1.0.0\android\src\main\kotlin\dev\steenbakker\mobile_scanner\MobileScanner.kt: (29, 1): Class 'MobileScanner' is not abstract and does not implement abstract member public abstract fun onRequestPermissionsResult(p0: Int, p1: Array<(out) String!>, p2: IntArray): Boolean defined in io.flutter.plugin.common.PluginRegistry.RequestPermissionsResultListener
e: D:\Development\SDK\Flutter\.pub-cache\hosted\pub.dartlang.org\mobile_scanner-1.0.0\android\src\main\kotlin\dev\steenbakker\mobile_scanner\MobileScanner.kt: (69, 5): 'onRequestPermissionsResult' overrides nothing
e: D:\Development\SDK\Flutter\.pub-cache\hosted\pub.dartlang.org\mobile_scanner-1.0.0\android\src\main\kotlin\dev\steenbakker\mobile_scanner\MobileScanner.kt: (70, 66): Type mismatch: inferred type is Array<CapturedType(out String)>? but Array<(out) String!> was expected
e: D:\Development\SDK\Flutter\.pub-cache\hosted\pub.dartlang.org\mobile_scanner-1.0.0\android\src\main\kotlin\dev\steenbakker\mobile_scanner\MobileScanner.kt: (70, 79): Type mismatch: inferred type is IntArray? but IntArray was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':mobile_scanner:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 56s
Exception: Gradle task assembleDebug failed with exit code 1
```

**Comment by quiquegarcia** [comment]
Confirmed

**Comment by MiniOcean404** [comment]
Just delete it manually

**Comment by Terranic** [comment]
> Just delete it manually

does not help. 
Even more:
> import 'package:mobile_scanner/mobile_scanner.dart';
result in "target uri not found"

**Comment by uSlashVlad** [comment]
Confirmed

**Comment by brim-borium** [comment]
Had the same issue and saw that the issue is the new method signature for `onRequestPermissionsResult ` in `MobileScanner.kt`. This issues has already been fixed on master with this commit [10915e492de4de7329c85fa2fd9e08931a6a7a80 ](https://github.com/juliansteenbakker/mobile_scanner/commit/10915e492de4de7329c85fa2fd9e08931a6a7a80)

@juliansteenbakker can you release a new version of the package ? This would fix the current issue with Flutter 3. :)

**Comment by utamori** [comment]
I was able to build using master as a dependency. Thanks.
`pubspec.yaml`
```yml
  mobile_scanner:
    git:
      url: https://github.com/juliansteenbakker/mobile_scanner.git
      ref: master
```

**Comment by juliansteenbakker** [comment]
Fixed in v1.1.0. Sorry for all the delay.

**Comment by brim-borium** [comment]
Thanks!
